### PR TITLE
chore: updates golangci-lint to `v1.52.2`

### PR DESCRIPTION
**Pull request by nilekhc**
**Reason for Change**:

**Issue Fixed**:

**Notes for Reviewers**:

### CODE REVIEW

File: pkg/plugin/server.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ type Config struct {
 }
 
 // New creates an instance of the KMS Service Server.
-func New(ctx context.Context, pc *Config) (*KeyManagementServiceServer, error) {
+func New(_ context.Context, pc *Config) (*KeyManagementServiceServer, error) {
```
Comment (aramase) [inline pkg/plugin/server.go:38]: If context is not used here, let's remove it!

**Review by aramase — CHANGES_REQUESTED** [review]
The integration tests failed because the credentials used expired. I've rotated it.

File: pkg/plugin/server.go (hunk 1 above)
Comment (nilekhc) [reply pkg/plugin/server.go:38]: I intend to figure out the use of context in https://github.com/Azure/kubernetes-kms/pull/210 hence did not remove it here.

Comment (aramase) [reply pkg/plugin/server.go:38]: Adding it back in #210 if it's used would be fine but given it's not used now, lets remove it.

**Review by aramase — APPROVED** [review]

**Pull request merged**